exchange/bidder.go

```diff
@@ -751,3 +744,40 @@ func hasShorterDurationThanTmax(ctx bidderTmaxContext, tmaxAdjustments TmaxAdjus
 	}
 	return false
 }
+
+func setRequestBody(req *adapters.RequestData, endpointCompression string) ([]byte, error) {
+	var requestBody []byte
+
+	switch strings.ToUpper(endpointCompression) {
+	case Gzip:
+		// Compress to GZIP
+		var b bytes.Buffer
+		w := gzipWriterPool.Get().(*gzip.Writer)
+		w.Reset(&b)
+		_, err := w.Write([]byte(requestBody))
+		if err != nil {
+			return nil, err
```

Comment:
You need to return the writer to the pool before exiting, otherwise there's a memory leak. Since you've extracted this logic to a method, I think a `defer` statement would work well.